eZ/Publish/Core/FieldType/Keyword/KeywordStorage.php

```diff
@@ -58,10 +58,20 @@ public function getFieldData(VersionInfo $versionInfo, Field $field, array $cont
      */
     public function deleteFieldData(VersionInfo $versionInfo, array $fieldIds, array $context)
     {
+        // If current version being asked to be deleted is not published, then don't delete keywords
+        // if there is some other version which is published (as keyword table is not versioned)
+        if ($versionInfo->status !== VersionInfo::STATUS_PUBLISHED &&
+            $versionInfo->contentInfo->isPublished
+        ) {
+            return false;
+        }
+
         $gateway = $this->getGateway($context);
         foreach ($fieldIds as $fieldId) {
             $gateway->deleteFieldData($fieldId);
         }
+
+        return true;
```

Comment:
Side note: `bool` is required by the `\eZ\Publish\SPI\FieldType\FieldStorage::deleteFieldData`. Method result is not used anywhere, so it worked before.